### DISCUSSION TITLE
How To Display Images in SLINT-based UI's?

### DISCUSSION THREAD

**Discussion by PackMeister** [slint-ui/slint#5904] [Q&A]
Is there a dedicated widget of some kind for displaying image files in SLINT UI's?

The image filter example below clearly shows images can be displayed but I was unable to find in [SLINT Reference Docs](https://releases.slint.dev/1.7.2/docs/slint/) the widget or other mechanism for how image files are to be displayed.

https://github.com/slint-ui/slint/tree/master/examples/imagefilter

**Comment by Vadoola** [comment]
The image filter example should be using the built in Image Element: https://releases.slint.dev/1.7.2/docs/slint/src/language/builtins/elements.html#image

**Reply by PackMeister** [reply]
Does the built-in Image Element require the selection and integration of [one of the three renderers mentioned here](https://releases.slint.dev/1.7.2/docs/slint/src/advanced/backends_and_renderers)?

So using SLINT's built-in Image Element in a desktop app should permit the app to access and display image files stored in some defined/indicated folder of the desktop OS...yes?

**Reply by Vadoola** [reply]
> Does the built-in Image Element require the selection and integration of [one of the three renderers mentioned here](https://releases.slint.dev/1.7.2/docs/slint/src/advanced/backends_and_renderers)?

Slint requires a renderer but it should use one by default as listed in that table, although you can override it. I would assume that yes the Image element would need to be supported by the renderer, and if it's not it won't work. I have never looked into this too closely, and it's possible it's supported by all of the available renderers. For example I know the Software Renderer doesn't support the Path Element, but in the link you provided it does say on the Software Renderer: "No image rotation or smooth scaling." Which implies the image element is supported.

> So using SLINT's built-in Image Element in a desktop app should permit the app to access and display image files stored in some defined/indicated folder of the desktop OS...yes?

Sure, I haven't looked closely at the Image Filter example for how they do it, but in my app [Tomotroid](https://github.com/Vadoola/Tomotroid/tree/main) I am using the image element to display SVGs. Some of them are icons and are statically set in the .slint ui files. Two of them are loaded in from disk in the Rust code, and can be updated at runtime (although in my app it's not reloading new images from disk, but updating the SVG images in memory and setting them to the image element in slint).
